src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -1 +1,275 @@
+open Core_kernel
+open Protocols.Coda_pow
 
+module type Inputs_intf = sig
+  module Time : Time_intf
+  module Proof : Proof_intf
+  module State_hash : Hash_intf
+  module Ledger_hash : Ledger_hash_intf
+  module Frozen_ledger_hash : Frozen_ledger_hash_intf
+    with type ledger_hash := Ledger_hash.t
+  module Ledger_builder_aux_hash : Ledger_builder_aux_hash_intf
+  module Ledger_builder_hash : Ledger_builder_hash_intf
+    with type ledger_hash := Ledger_hash.t
+     and type ledger_builder_aux_hash := Ledger_builder_aux_hash.t
+  module Ledger_builder_diff : Ledger_builder_diff_intf
+
+  module Blockchain_state : Blockchain_state_intf
+    with type ledger_builder_hash := Ledger_builder_hash.t
+     and type frozen_ledger_hash := Frozen_ledger_hash.t
+     and type time := Time.t
+
+  module Consensus_state : Consensus_state_intf
+
+  module Protocol_state : Protocol_state_intf
+    with type state_hash := State_hash.t
+     and type blockchain_state := Blockchain_state.value
+     and type consensus_state := Consensus_state.value
+
+  module External_transition : External_transition_intf
+    with type protocol_state := Protocol_state.value
+     and type protocol_state_proof := Proof.t
+     and type ledger_builder_diff := Ledger_builder_diff.t
+
+  module Merkle_mask : sig
+    type t
+
+    val root : t -> Ledger_hash.t
+    val derive : t -> t
+    val merge : t -> t -> unit
+    val apply : t -> Ledger_builder_diff.t -> unit
+  end
```

Comment:
see interfaces: https://github.com/CodaProtocol/coda/blob/master/src/lib/merkle_mask/masking_merkle_tree_intf.ml

Reply:
If you want an interface like this, please change the underlying merkle mask code too

Reply:
Also right now the merkle_mask abstraction is on a ledger not a ledger-builder

Reply:
Oh, I didn't know the interface was in yet. Well, the derive and merge should still exist in some capacity. Those were described in the original RFC. The apply was just a shorthand for iterating over the ledger builder diff and applying the transactions one by one. Those need to be bulked for the merkle mask interface anyway, as per the RFC.